kvx/src/implementations/disk.rs

```diff
@@ -348,63 +346,33 @@ impl PathBufExt for PathBuf {
 
 #[derive(Debug)]
 struct FileLock {
-    file: File,
-    lock_path: PathBuf,
+    lock: fd_lock::RwLock<File>,
 }
 
 impl FileLock {
-    const POLL_LOCK_INTERVAL: Duration = Duration::from_millis(10);
-
-    pub fn lock(path: impl AsRef<Path>) -> Result<Self> {
+    fn create(path: impl AsRef<Path>) -> Result<Self> {
```

Comment:
I wouldn’t use the `imp AsRef<_>` generics for the path argument on a private function – not that it matters too much.

Reply:
fixed